src/main/java/BuyStockUseCase/PortfolioPresenter.java

```diff
@@ -39,6 +42,12 @@ public void onBack() {
     }
     public void onSearch() {
         String symbol = view.getSearchField();
+        try{
+            new ViewStockController(symbol).stockIsValid();
+        }catch (Exception e){
+            JOptionPane.showMessageDialog(null, "Invalid stock symbol: "+symbol);
```

Comment:
I think you could refactor your code around here and place it in a different file

Reply:
yea this code should be put into a function in the GUI class called presentInvalidStockMessage and the presenter should call it through the interface the GUI implements

Reply:
Thanks for the feedback, I have moved some logic into a function in the GUI.